### PR TITLE
move to github actions

### PR DESCRIPTION
**Pull request by ahartikainen**
#### Submission Checklist

- [x] Run unit tests
- [x] Declare copyright holder and open-source license: see below

#### Summary

Test on Github Actions (Linux, macOS and Windows) with Python (3.6-3.9).
Automatically upload wheels to PyPI if release/tag.

#### Copyright and Licensing

Please list the copyright holder for the work you are submitting (this will be you or your assignee, such as a university or company):

Ari Hartikainen

By submitting this pull request, the copyright holder is agreeing to license the submitted work under the following licenses:
- Code: BSD 3-clause (https://opensource.org/licenses/BSD-3-Clause)

### CODE REVIEW

**Comment by ahartikainen** [comment]
@serban-nicusor-toptal or @mitzimorris can you create a pypi token and put it under github secrets

secrets.PYPI_USERNAME = __token__
secrets.PYPI_PASSWORD = token-here

**Comment by ahartikainen** [comment]
This is now ready. Let's first fix the tests and then rebase and check if everything works.

**Comment by serban-nicusor-toptal** [comment]
Hey, there is already `PYPI_PASSWORD`, I think it's a token generated by @mitzimorris sadly I don't know the username.

**Comment by ahartikainen** [comment]
Sure, sounds ok

**Comment by mitzimorris** [comment]
pypi account username is CmdStanPy

**Comment by mitzimorris** [comment]
@ahartikainen  - what's your pypi name?

**Comment by serban-nicusor-toptal** [comment]
`PYPI_USERNAME` added to secrets, please let me know if it works!

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+          pip install "$whl"
+
+      - name: Show libraries
+        run: |
+          python -m pip freeze
+
+      - name: Install CmdStan (Linux, macOS)
+        if: matrix.os != 'windows-latest'
+        run: |
+          python -m cmdstanpy.install_cmdstan --progress
```
Comment (mitzimorris) [inline .github/workflows/main.yml:64]: is flag --progress necessary?

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+          python scripts/clean_examples.py
+
+      - name: Install CmdStan (Windows)
+        if: matrix.os == 'windows-latest'
+        run: |
+          python -m cmdstanpy.install_cmdstan --progress --compiler
```
Comment (mitzimorris) [inline .github/workflows/main.yml:70]: ditto - --progress necessary or just nice to have?

**Review by mitzimorris — APPROVED** [review]
LGTM!

File: .github/workflows/main.yml (hunk 1 above)
Comment (ahartikainen) [reply .github/workflows/main.yml:64]: Just nice to have. It is so fast in CI that in normal case, devs don't need to care about it.

**Comment by ahartikainen** [comment]
Ok, I removed sample sigfigs.

I don't think it actually did not test the correct thing.

We can fix this in another PR where we manually parse sample csv and check number len in str format. str -> float does not work.

**Comment by ahartikainen** [comment]
https://github.com/stan-dev/cmdstanpy/issues/330

File: cmdstanpy/utils.py [hunk 3]
```diff
@@ -1,36 +1,31 @@
...
+import tempfile
+from collections import OrderedDict
+from collections.abc import Sequence
+from numbers import Integral, Real
+from typing import Dict, List, TextIO, Tuple, Union
 
 import numpy as np
 import pandas as pd
-
+import ujson as json
```
Comment (mitzimorris) [inline cmdstanpy/utils.py:21]: thanks for cleaning up all of the imports!

**Review by mitzimorris — APPROVED** [review]
this looks GREAT!

**Pull request merged**